src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -452,7 +452,42 @@ ChannelsStatusResponse getChannelsStatus(
    * @param blobs list of uploaded blobs
    */
   void registerBlobs(List<BlobMetadata> blobs) {
-    this.registerBlobs(blobs, 0);
+    for (List<BlobMetadata> blobBatch : partitionBlobListForRegistrationRequest(blobs)) {
+      this.registerBlobs(blobBatch, 0);
+    }
+  }
+
+  /**
+   * Partition the collection of blobs into sub-lists, so that the total number of chunks in each
+   * sublist does not exceed the max allowed number of chunks in one registration request.
+   */
+  List<List<BlobMetadata>> partitionBlobListForRegistrationRequest(List<BlobMetadata> blobs) {
+    List<List<BlobMetadata>> result = new ArrayList<>();
+    List<BlobMetadata> currentBatch = new ArrayList<>();
+    int chunksInCurrentBatch = 0;
+
+    for (BlobMetadata blob : blobs) {
+      if (chunksInCurrentBatch + blob.getChunks().size()
+          > parameterProvider.getMaxChunksInRegistrationRequest()) {
+        // Newly added BDEC file would exceed the max number of chunks in a single registration
+        // request. We put chunks collected so far into the result list and create a new batch with
+        // the current blob
+        result.add(currentBatch);
```

Comment:
nit: This probably can't happen as long as `maxChunksInBlob` parameter is less than `getMaxChunksInRegistrationRequest` parameter but if it is not we will add a `currentBatch` that is empty to the `result` list.

Reply:
Good point, I added validation that `maxChunksInBlob` is always less than `getMaxChunksInRegistrationRequest`